pallets/did/src/did_details.rs

```diff
@@ -340,19 +343,34 @@ impl<T: Config> DidDetails<T> {
 	pub fn update_deposit(&mut self, did_subject: &DidIdentifierOf<T>) -> Result<(), DispatchError> {
 		let new_required_deposit = self.calculate_deposit(did_subject);
 
+		let is_key_migrated = <T as Config>::MigrationManager::is_key_migrated(Did::<T>::hashed_key_for(did_subject));
+
 		match new_required_deposit.cmp(&self.deposit.amount) {
 			Ordering::Greater => {
 				let deposit_to_reserve = new_required_deposit.saturating_sub(self.deposit.amount);
+
+				if !is_key_migrated {
+					<T as Config>::MigrationManager::release_reserved_deposit(
+						&self.deposit.owner,
+						&self.deposit.amount,
+					);
+					<T as Config>::MigrationManager::exclude_key_from_migration(Did::<T>::hashed_key_for(did_subject));
+				}
 				DidDepositCollector::<T>::create_deposit(self.deposit.clone().owner, deposit_to_reserve)?;
 				self.deposit.amount = self.deposit.amount.saturating_add(deposit_to_reserve);
```

Comment:
Does this not release the full deposit in case it's not migrated and after that only reserve the new additional deposit?

Reply:
Sorry – forgot to mark this PR back to a draft.  Updated in: [3cfd1de](https://github.com/KILTprotocol/kilt-node/pull/540/commits/3cfd1de4ef886079e8e8e712615b481b98fda8bb)